include/yui/types.h

```diff
@@ -119,24 +125,74 @@ inline vec2 operator/(const vec2& lhs, float rhs) noexcept {
  * @brief   Internal vec4 type
  */
 struct vec4 {
```

Comment:
The `vec4` shouldn't be in this repository. Similarly `vec2` and `vec3`. Implement the `vec<N>` in `nova` like the `std::tuple`.